src/item.cpp

```diff
@@ -3481,31 +3475,28 @@ bool item::is_ammo_container() const
     return !is_magazine() && !contents.empty() && contents.front().is_ammo();
 }
 
-bool item::is_weap() const
+bool item::is_melee() const
 {
-    if( is_null() )
-        return false;
-
-    if (is_gun() || is_food() || is_ammo() || is_food_container() || is_armor() ||
-            is_book() || is_tool() || is_bionic() || is_gunmod())
-        return false;
-    return (type->melee_dam > 7 || type->melee_cut > 5);
+    for( auto idx = DT_NULL + 1; idx != NUM_DT; ++idx ) {
+        if( is_melee( static_cast<damage_type>( idx ) ) ) {
+            return true;
+        }
+    }
+    return false;
 }
 
-bool item::is_bashing_weapon() const
+bool item::is_melee( damage_type dt ) const
 {
-    if( is_null() )
+    if( is_null() || is_gun() || is_gunmod() || is_food() || is_ammo() ||
+        is_food_container() || is_armor() || is_book() || is_bionic() ) {
```

Comment:
This whole check is needlessly restricting.
Consider the cases:
* Someone grabs an alien leg (`is_food() == true`) and bashes stuff with it
* Gun is used to bash/stab something
* Someone mods in gasoline as edible, chainsaw suddenly stops being a weapon when loaded
* Bible-thumping with a very large bible

`is_null` would make sense (but be redundant due to low damage) here if you didn't add the unarmed check below - if unarmed weapons are weapons, then pure unarmed should count too.